weasyl/controllers/content.py

```diff
@@ -30,58 +30,71 @@ def submit_(request):
+    baseURL = request.params.get('baseURL', '')
+    imageURL = request.params.get('imageURL', '')
+    tags = request.params.get('tags', '').split(',')
+    if baseURL:
+        imageURL = urlparse.urljoin(baseURL, imageURL)
 
     return Response(define.webpage(request.userid, "submit/visual.html", [
         # Folders
         folder.select_list(request.userid, "drop/all"),
         # Subtypes
         [i for i in macro.MACRO_SUBCAT_LIST if 1000 <= i[0] < 2000],
         profile.get_user_ratings(request.userid),
-        form,
+        title, imageURL, tags, description
     ], title="Visual Artwork"))
 
 
 @login_required
 @token_checked
 def submit_visual_post_(request):
-    form = request.web_input(submitfile="", thumbfile="", title="", folderid="",
-                             subtype="", rating="", friends="", critique="", content="",
-                             tags="", imageURL="")
-
-    tags = searchtag.parse_tags(form.tags)
+    tags = searchtag.parse_tags(request.params.get('tags', ''))
 
     if not define.config_read_bool("allow_submit"):
         raise WeasylError("FeatureDisabled")
 
     if not define.is_vouched_for(request.userid):
         raise WeasylError("vouchRequired")
 
-    rating = ratings.CODE_MAP.get(define.get_int(form.rating))
+    rating = ratings.CODE_MAP.get(define.get_int(request.params.get('rating', '')))
     if not rating:
         raise WeasylError("ratingInvalid")
 
     s = orm.Submission()
-    s.title = form.title
+    s.title = request.params.get('title', '')
     s.rating = rating
-    s.content = form.content
-    s.folderid = define.get_int(form.folderid) or None
-    s.subtype = define.get_int(form.subtype)
+    s.content = request.params.get('content', '')
+    s.folderid = define.get_int(request.params.get('folderid', '')) or None
+    s.subtype = define.get_int(request.params.get('subtype', ''))
     s.submitter_ip_address = request.client_addr
     s.submitter_user_agent_id = get_user_agent_id(ua_string=request.user_agent)
 
-    submitid = submission.create_visual(
-        request.userid, s, friends_only=form.friends, tags=tags,
-        imageURL=form.imageURL, thumbfile=form.thumbfile, submitfile=form.submitfile,
-        critique=form.critique, create_notifications=('nonotification' not in form))
+    submitfile = request.params.get('submitfile', u'')
+    if submitfile != u'':
+        submitfile = submitfile.file.read()
+    thumbfile = request.params.get('thumbfile', u'')
+    if thumbfile != u'':
+        thumbfile = thumbfile.file.read()
 
-    if 'customthumb' in form:
+    submitid = submission.create_visual(
+        request.userid,
+        s,
+        friends_only=request.params.get('friends', ''),
+        tags=tags,
+        imageURL=request.params.get('imageURL', ''),
+        thumbfile=thumbfile,
+        submitfile=submitfile,
+        critique=request.params.get('critique', False),
+        create_notifications=(request.params.get('nonotification', False) is not True)
```

Comment:
I don’t think that works.

```suggestion
        create_notifications=('nonotification' not in request.params)
```